lib/block_view.go

```diff
@@ -4160,6 +4169,56 @@ func (bav *UtxoView) ConnectBlock(
 		}
 	}
 
+	// If we're past the PoS cutover, we need to track which validators were active.
+	if blockHeight >= uint64(bav.Params.ForkHeights.ProofOfStake2ConsensusCutoverBlockHeight) {
+		// Get the active validators for the block.
+		var signersList *bitset.Bitset
+		if !desoBlock.Header.ValidatorsVoteQC.isEmpty() {
+			signersList = desoBlock.Header.ValidatorsVoteQC.ValidatorsVoteAggregatedSignature.SignersList
+		} else {
+			signersList = desoBlock.Header.ValidatorsTimeoutAggregateQC.ValidatorsTimeoutAggregatedSignature.SignersList
+		}
+		allSnapshotValidators, err := bav.GetAllSnapshotValidatorSetEntriesByStake()
+		if err != nil {
+			return nil, errors.Wrapf(err, "ConnectBlock: error getting all snapshot validator set entries by stake")
+		}
+		currentEpochNumber, err := bav.GetCurrentEpochNumber()
+		if err != nil {
+			return nil, errors.Wrapf(err, "ConnectBlock: error getting current epoch number")
+		}
+		// TODO: Is it safe to assume that the leader is included in the signers list?
+		// If not, we should explicitly add them here since they were "active".
+		for ii, validator := range allSnapshotValidators {
+			// Skip validators who didn't sign
+			if !signersList.Get(ii) {
+				continue
+			}
+			// Get the current validator entry
+			validatorEntry, err := bav.GetValidatorByPKID(validator.ValidatorPKID)
+			if err != nil {
+				return nil, errors.Wrapf(err, "ConnectBlock: error getting validator by PKID")
+			}
+			// TODO: what do we want to do for jailed validators? I'm assuming we shouldn't hit them here.
+			if validatorEntry.Status() == ValidatorStatusJailed {
+				glog.Errorf("ConnectBlock: Validator %v is jailed but signed the block",
+					PkToString(validatorEntry.ValidatorPKID.ToBytes(), bav.Params))
+				continue
+			}
+			if validatorEntry.LastActiveAtEpochNumber != currentEpochNumber {
+				// TODO: do we want to make a PrevValidatorEntries attribute for
+				// the utxo operation struct?
```

Comment:
Up to you. For completeness, it seems like capturing the previous state of the validator seems to be the proper thing to do. We know that each validator will only appear once per epoch in the UtxoOps here, so IMO this is safe.
```suggestion
```

Reply:
@tholonious - my question here is not whether or not we want to track it, but if we should add a `PrevValidatorEntries` slice to UtxoOperation to capture all of these changes in a single utxo operation. I think it's fine to leave it as is for now.